src/ui/DynamicFacet/DynamicFacetValues/DynamicFacetRangeValueUtils.ts

```diff
@@ -1,50 +1,73 @@
+import * as Globalize from 'globalize';
 import { RangeType, IRangeValue, RangeEndScope } from '../../../rest/RangeValue';
 import { DynamicFacetRangeValueFormat } from '../DynamicFacetRange';
 import { NumberUtils } from '../../../utils/NumberUtils';
-import * as Globalize from 'globalize';
 import { isNull } from 'util';
+import { DateUtils } from '../../../utils/DateUtils';
 
 export class DynamicFacetRangeValueUtils {
-  public static formatValue(value: RangeType, valueFormat: DynamicFacetRangeValueFormat) {
+  public static formatDisplayValue(value: RangeType, valueFormat: DynamicFacetRangeValueFormat) {
     switch (valueFormat) {
       case DynamicFacetRangeValueFormat.number:
-        return this.formatNumberValue(<number>value);
-      // TODO: Manage more value formats
+        return this.formatNumberDisplayValue(<number>value);
+      case DynamicFacetRangeValueFormat.date:
+        return this.formatDateDisplayValue(value);
       default:
         return `${value}`;
     }
   }
 
-  private static formatNumberValue(value: number): string {
+  private static formatNumberDisplayValue(value: number): string {
     const numberOfDecimals = NumberUtils.countDecimals(value);
     return Globalize.format(value, `n${numberOfDecimals}`);
   }
 
-  public static formatRangeValue(range: IRangeValue, valueFormat: DynamicFacetRangeValueFormat, valueSeparator: string) {
-    const formattedStart = this.formatValue(range.start, valueFormat);
-    const formattedEnd = this.formatValue(range.end, valueFormat);
+  private static formatDateDisplayValue(value: RangeType): string {
+    return DateUtils.dateToString(this.dateFromRangeTypeValue(value));
+  }
+
+  private static dateFromRangeTypeValue(value: RangeType) {
```

Comment:
Had to find a way to make Typescript manage well the Date vs RangeType types, this method is necessary to prevent any types issues.

Reply:
I think you could leverage type guards in TypeScript to tell the compiler that the value `is` of type `Date` or `Number` when you use it.

https://www.typescriptlang.org/docs/handbook/advanced-types.html#using-type-predicates